websockify.yaml

```
@@ -31,18 +37,55 @@ pipeline:
       tag: v${{package.version}}
       expected-commit: 99f83ca08390dc876b1b3580c210abea5b9f4edd
 
-  - name: Python Build
-    uses: python/build-wheel
+subpackages:
+  - range: py-versions
+    name: py${{range.key}}-${{vars.pypi-package}}
+    description: python${{range.key}} version of ${{vars.pypi-package}}
+    dependencies:
+      provider-priority: ${{range.value}}
+      runtime:
+        - py${{range.key}}-numpy
+        - py${{range.key}}-jwcrypto
+        - py${{range.key}}-requests
+        - py${{range.key}}-simplejson
+    pipeline:
+      - uses: py/pip-build-install
+        with:
+          python: python${{range.key}}
+      - name: "move usr/bin executables for -bin"
+        runs: |
+          mkdir -p ./cleanup/${{range.key}}/
+          mv ${{targets.contextdir}}/usr/bin ./cleanup/${{range.key}}/
+      - uses: strip
+    test:
+      pipeline:
+        - uses: python/import
+          with:
+            python: python${{range.key}}
+            import: ${{vars.import}}
 
-  - uses: strip
+  - range: py-versions
+    name: py${{range.key}}-${{vars.pypi-package}}-bin
+    description: Executable binaries for ${{vars.pypi-package}} installed for python${{range.key}}
+    dependencies:
+      runtime:
+        - py${{range.key}}-${{vars.pypi-package}}
+      provides:
+        - py3-${{vars.pypi-package}}-bin
+        - websockify
```

```suggestion
        - py3-${{vars.pypi-package}}
```
I think we should be using this package now as runtime dependency after migrating to multiversion support. websockify is empty [package](https://apk.chaindag.dev/https/apk.cgr.dev/wolfi-presubmit/9125a36bf69507ddb282df7987f7d1e869aeb356/aarch64/websockify-0.12.0-r3.apk@etag:7621302313b3155aa50af2af463a2ea7/) via this commit.

---

I added a provides for `websockify` because of all the potential images and packages that could depend on the old `websockify` package name. Is it not practice to provide the old names for backwards compatibility?

---

thank you for the explanation, this looks good to me. Can we also consider adding `py3-${{vars.pypi-package}}` together with this so that we have it synced with other examples and if someone wants to pull in the latest then they can do `apk add py3-websockify` or specify the same as runtime dep.

I think we also don't need to check on dependents `docker-selenium` and `novnc` to check if they're building correctly after this migration because they're already building with latest python which is inline with this. most likely they'll build fine. 💭

---

Aaah, good idea with adding that one, doing it! Thank you!